internal/reflect/into.go

```diff
@@ -22,7 +23,7 @@ import (
 // in the tftypes.Value must have a corresponding property in the struct. Into
 // will be called for each struct field. Slices will have Into called for each
 // element.
-func Into(ctx context.Context, typ attr.Type, val tftypes.Value, target interface{}, opts Options) diag.Diagnostics {
+func Into(ctx context.Context, typ attr.Type, val tftypes.Value, target interface{}, opts Options, path path.Path) diag.Diagnostics {
```

Comment:
Now that the `path.Path` information is available in this function, we should replace the `diags.AddError()` call below with `diags.AddAttributeError()` and consider adding `Path: path.String()\nError: err.Error()` to the details.